clickstream-event-visualiser/src/main/java/clickstream/event_visualiser/CSEVEventObserver.kt

```diff
@@ -0,0 +1,18 @@
+package clickstream.event_visualiser
+
+import clickstream.interceptor.CSEventInterceptor
+import clickstream.interceptor.CSInterceptedEvent
+
+
+/**
+ * Observer class that manages callback and emits event updates from [CSEventInterceptor]
+ *
+ * */
+public interface CSEVEventObserver {
+
+    public fun addCallback(callback: (List<CSInterceptedEvent>) -> Unit)
+
+    public fun removeCallback(callback: (List<CSInterceptedEvent>) -> Unit)
+
+    public fun onEventChanged(events: List<CSInterceptedEvent>)
```

Comment:
Should we rename `onEventChanged` to something which depicts a flow of events rather than this? This feels more like a change of state.